app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -288,6 +337,25 @@ class BrowserFragment : BaseBrowserFragment(), BackHandler {
     override fun onSessionSelected(session: Session) {
         super.onSessionSelected(session)
         quickActionSheetSessionObserver?.updateBookmarkState(session)
+        updateTrackingProtectionUIState(session)
+    }
+
+    private fun updateTrackingProtectionUIState(session: Session) {
+        exceptionJob?.cancel()
+        exceptionJob = lifecycleScope.launch(IO) {
+            val host = try {
+                URL(session.url).host
+            } catch (e: MalformedURLException) {
+                session.url
+            }
+            val trackingProtectionException = ExceptionDomains.load(context!!).contains(host)
+            withContext(Main) {
```

Comment:
We can take out this, to have a consistent experience as it will be handled when next week when AC will provide the exception handling API
https://github.com/mozilla-mobile/android-components/issues/3264

Reply:
Added a fix for it